src/components/Ideas/CompIdeaDetail.js

```diff
@@ -0,0 +1,45 @@
+import React, { useContext, useEffect, useState } from "react"
+import { IdeaContext } from "./IdeaProvider"
+import { useParams, useHistory } from "react-router-dom"
+import "./Idea.css"
+
+export const CompIdeaDetail = (props) => {
+    const { ideas, getCompleteIdeas, deleteIdea } = useContext (IdeaContext)
+    const [ idea, setIdea ] = useState({ artist: {} })
+
+    const { ideaId } = useParams()
+
+    useEffect(() => {
+        if(props.idea){
+            setIdea(props.idea)
+        } else {
+            const thisIdea = idea.find(i => i.id === parseInt(ideaId)) || { artist: {} }
+            setIdea(thisIdea)
+        }
+    }, [idea, ideas])
+
+    useEffect(() => {getCompleteIdeas()}, [])
+
+    const handleDeleteIdea = () => {
+        deleteIdea(idea.id)
+        .then(() => {
+            history.push("/ideas")
+        })
+    }
+
+    const history = useHistory()
+
+    return (
+    <section className="idea">
```

Comment:
This looks to be an idea card. So this could be turned into another component. SInce this card could possibly be used somewhere else.